src/gui/utils.cpp

```diff
@@ -76,35 +75,6 @@ qreal Utils::Gui::screenScalingFactor(const QWidget *widget)
 #endif // Q_OS_WIN
 }
 
-QPixmap Utils::Gui::scaledPixmap(const QIcon &icon, const QWidget *widget, const int height)
```

Comment:
I haven't been following closely to this PR, so pardon if it doesn't make sense in advance.
I think merging these functions within existing uithememanager methods is a bad idea, those are image scaling functions, it should be preserved in it own form (private functions in uithememanager is acceptable for me, and making it public when required). See how you load `mascot.png`, spilling all the unnecessary details of image scaling details into it.

Reply:
>I haven't been following closely to this PR

UIThemeManager should provide icon (or pixmap) by its ID.
The client code (of UIThemeManager) should know nothing about theme details (e.g. real file types etc.).
In some places we need to get scaled pixmap. Currently we have two helpers to scale pixmap (depending on its file type). But we can't proceed to use this way considering the above. So I suggested moving this logic inside the UIThemeManager class. Otherwise we can leave only one helper that will scale existing pixmap (not so optimal since we can't use SVG optimization).

Reply:
I think this can work (a careless sketch up):
```c++
QPixmap UIThemeManager::getScaledPixmap(const QString &iconId, const QWidget *widget, int baseHeight) const
{
    const QString iconPath = UIThemeManager::instance()->getIconPath(iconId);
    return Utils::Gui::scaledPixmap(iconPath, widget, baseHeight);  // IMO this function has nothing to do with theme manager
}

QPixmap Utils::Gui::scaledPixmap(const QString &iconPath, const QWidget *widget, const int baseHeight)
{
    const int scaledHeight = baseHeight * Utils::Gui::screenScalingFactor(widget);
    const QString normalizedKey = iconPath + '@' + QString::number(scaledHeight);

    QPixmap pm;
    QPixmapCache cache;
    if (!cache.find(normalizedKey, &pm)) {
        pm = QIcon(iconPath).pixmap(scaledHeight);     
        cache.insert(normalizedKey, pm);
    }
    return pm;
}
```

Reply:
>I think this can work (a careless sketch up):

I think, Yes.
But UIThemeManager instance shouldn't call itself indirectly:
```c++
QPixmap UIThemeManager::getScaledPixmap(const QString &iconId, const QWidget *widget, int baseHeight) const
{
    const QString iconPath = getIconPath(iconId);
    return Utils::Gui::scaledPixmap(iconPath, widget, baseHeight);
}
```

Also I wouldn't have `widget` as param but `scalingFactor` to be more abstract.

Reply:
now?

Reply:
Now I have reset the changes inside `utils.cpp`.  I don't do anything with `utils.*`.  Changed `getPixmap` to `getScaledPixmap` which calls `Utils::Gui::scaledPixmap` as you guys said.